optax/contrib/cocob.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2019 DeepMind Technologies Limited. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from typing import NamedTuple
+
+import jax
+import jax.numpy as jnp
+
+from optax._src import base
+
+
+class COCOBState(NamedTuple):
+  """State for COntinuous COin Betting."""
+  init_particles: base.Updates
+  cumulative_gradients: base.Updates
+  scale: base.Updates
+  subgradients: base.Updates
+  reward: base.Updates
+
+
+def cocob(alpha: float = 100, eps: float = 1e-8) -> base.GradientTransformation:
+  """Rescale updates according to the parameter-free backproagating version of the COntinuous COin Betting algorithm.
```

Comment:
Line too long.

"""Rescale updates according to the COntinuous COin Betting algorithm."""